.github/workflows/deploy.yml

```diff
@@ -0,0 +1,90 @@
+        if: github.ref == 'refs/heads/master'
+        uses: elgohr/Publish-Docker-Github-Action@master
+        with:
+          name: demo-storefrontcloud-io/vue-storefront-api:${{ github.sha }}
+          registry: registry.storefrontcloud.io
+          username: ${{ secrets.CLOUD_USERNAME }}
+          password: ${{ secrets.CLOUD_PASSWORD }}
+          dockerfile: dev/docker/Dockerfile
+          buildoptions: "--compress"
+      
+      # Build for test
+      - name: Build and publish test docker image
+        if: github.ref == 'refs/heads/develop'
+        uses: elgohr/Publish-Docker-Github-Action@master
+        with:
+          name: test-storefrontcloud-io/vue-storefront-api:${{ github.sha }}
+          registry: registry.storefrontcloud.io
+          username: ${{ secrets.CLOUD_USERNAME }}
+          password: ${{ secrets.CLOUD_PASSWORD }}
+          dockerfile: dev/docker/Dockerfile
+          buildoptions: "--compress"
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: chrnorm/deployment-action@releases/v1
+        name: Create GitHub deployment for ${GITHUB_REF##*/}
+        if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/develop'
+        id: deployment
+        with:
+          token: "${{ github.token }}"
+          target_url: https://${GITHUB_REF##*/}.europe-west1.gcp.storefrontcloud.io
+          environment: production
+          initial_status: in_progress
+
+      # Deploy on test/demo depending on the branch that triggered the workflow
+      - name: Deploy on ${GITHUB_REF##*/}.europe-west1.gcp.storefrontcloud.io
+        if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/develop'
+        run: |
+          if curl -s -H 'X-User-Id: ${{ secrets.CLOUD_USERNAME }}'  -H 'X-Api-Key: ${{ secrets.CLOUD_PASSWORD }}' -H 'Content-Type: application/json' -X POST -d '
+          {
+              "code":"${GITHUB_REF##*/}",
+              "region":"europe-west1.gcp",
+              "frontContainerVersion":"${{ github.sha }}"
+          }' https://farmer.storefrontcloud.io/instances | grep -q '{"code":200,"result":"Instance updated!"}'; then
+            echo "Instance updated"
+          else
+            echo "Something went wrong during the update process..."
+            exit 1
+          fi
+
+      - name: Update deployment status (success)
+        if: success() && github.ref == 'refs/heads/master' || success() && github.ref == 'refs/heads/develop'
+        uses: chrnorm/deployment-status@releases/v1
+        with:
+          token: "${{ github.token }}"
+          target_url: https://${GITHUB_REF##*/}.europe-west1.gcp.storefrontcloud.io
+          state: "success"
+          description: Congratulations! The deploy is done.
+          deployment_id: ${{ steps.deployment.outputs.deployment_id }}
+      - name: Update deployment status (failure)
+        if: failure() && github.ref == 'refs/heads/master' || failure() && github.ref == 'refs/heads/develop'
```

Comment:
This if has to many statements, this job is always running for one of those branches because you have
```
on:
  push:
    branches:
      - master
      - develop
      
```